### PR TITLE
Invite Collaborators in Batch 

### PR DESCRIPTION
**Pull request by UkcaGreen**
`invite collaborators` and `join request` buttons are added to the project details page.

- By using `join request` button user can make a request to join the related project.
- By using `invite collaborators`  button user can invite other users to the related project. To make user be able to select users a modal is created. So that user can search and select multiple users to invite. When user click to the invite button, invitation is sent all the selected users.

Example Screenshots:
resim

### CODE REVIEW

**Comment by aliramazanmert** [comment]
I think this branch is way too behind from the frontend branch, we added join request button to project detail page but we could use invite collaborators part. You should first merge the latest version of frontend otherwise this pull request is not safe to merge.

**Review by aliramazanmert — DISMISSED** [review]
Need to catch up with the latest frontend version first, then we can add only the invite collaborators modal part

**Comment by UkcaGreen** [comment]
> Need to catch up with the latest frontend version first, then we can add only the invite collaborators modal part

Okey working on syncronization of this feature.

**Comment by UkcaGreen** [comment]
Synchronization is completed:
- I ajdusted the project so that the new `batch invite feature` is working smoothly.
- Related api function called `sendBatchInviteRequest` is created. 
- There were some issues due to brach being out dated, in almost all cases uptodate parts of the `frontend branch` is preffered over `frontend-notification` branch.

I believe branch is ready to be merged.

File: frontend/src/pages/Profile/index.jsx [hunk 1]
```diff
@@ -69,7 +68,7 @@ const Profile = () => {
       dispatch(scrolledToProjectsAction());
     }
     // eslint-disable-next-line
-  }, []);
+  }, [id]);
```
Comment (cemreefe) [inline frontend/src/pages/Profile/index.jsx:71]: What does this one do?

**Comment by cemreefe** [comment]
We have tested this branch with @UkcaGreen. I'm approving the merge.

**Review by cemreefe — APPROVED** [review]

**Pull request merged**